content/blog/2022-03-whats-new-in-axum-0-5.md

```diff
@@ -0,0 +1,298 @@
+---
+date: "2022-04-31"
+title: "What's new in axum 0.5"
+description: "March 31, 2022"
+---
+
+Today, we're happy to announce [axum] version 0.5. axum is an ergonomic and
+modular web framework built with tokio, tower, and hyper.
+
+0.5 contains lots of new features and I'd like highlight a few of them here.
+
+## The new [`IntoResponseParts`] trait
+
+axum has always supported building responses by composing individual parts:
+
+```rust
+use axum::{
+    Json,
+    response::IntoResponse,
+    http::{StatusCode, HeaderMap},
+};
+use serde_json::json;
+
+// returns a JSON response
+async fn json() -> impl IntoResponse {
+    Json(json!({ ... }))
+}
+
+// returns a JSON response with a `201 Created` status code and
+// a custom header
+async fn json_with_status_and_header() -> impl IntoResponse {
+    let mut headers = HeaderMap::new();
+    headers.insert("x-foo", "custom".parse().unwrap());
+
+    (StatusCode::CREATED, headers, Json(json!({})))
+}
+```
+
+However you couldn't easily provide your own custom response parts. axum had to
+specifically allow `HeaderMap` to be included in responses and you couldn't
+extend this system with your own types.
+
+The new `IntoResponseParts` trait fixes that!
```

Comment:
nit, take it or leave it: might be cool if this was a docs link?

Reply:
Yep! Will do when its shipped.